src/dhtnode/storage/StorageEngine.d

```diff
@@ -192,6 +192,21 @@ public class StorageEngine : IStorageEngine
     }
 
 
+    /***************************************************************************
+
+        Ditto
+
+    ***************************************************************************/
+
+    public typeof(this) put ( hash_t key, cstring value,
+        bool trigger_listeners = true )
+    {
+        char[hash_t.sizeof * 2] key_str;
+        Hash.toHexString(key, key_str);
```

Comment:
Hm, so the storage engine internally uses `hash_t`, but then you here convert the entire thing to `cstring` and then you call `this.put` which will convert it back to `hash_t`. Wouldn't it be better if we would move the implementation here and provide `cstring` methods as adapters to `hash_t`?

Reply:
Sigh... This is already done, just the order of commits is messed up.

Reply:
(in GitHub)